### DISCUSSION TITLE
Correctly setting `search_path` to include `"$user"`

### DISCUSSION THREAD

**Discussion by gaganawhad** [jeremyevans/sequel#1917] [General]
Hello, 

I am trying to set my search path to include the schema `"$user"`.  When, in `database.yml` I set it as so: 

```
search_path: $user, public, heroku_ext
```

My database connection sets the `search_path` correctly, but on running `db:migrate`, `schema.rb` gets the following SQL to set `search_path`: 

```
"SET search_path TO $user, public, heroku_ext"
```
It's missing the quotes for `$user`. 

Conversely, if I set `search_path` in `database.yml` to 

```
search_path: "\"$user\", public, heroku_ext"
```
I get the correct SQL in `schema.rb` like so: 

```
"SET search_path TO \"$user\", public, heroku_ext"
```
but my database connection shows `search_path` like this: 

```
 "\"\"\"$user\"\"\", public, heroku_ext"
```

How can I set search path to use `"$user"` correctly?

**Comment by jeremyevans** [comment]
Sequel by default does not use yaml files (outside of bin/sequel), nor does it create `schema.rb` or have `db:migrate` rake tasks.  Are you using some tool that integrates Sequel and Rails?  If so, you should probably raise an issue with that tool.

Sequel handles this correctly if you pass the search_path Database option, since it treats each option as a quoted identifier.  Here's an example using a connection URL:

```
bin/sequel -E 'postgres:///?search_path=$user,public,heroku_ext'
# SET search_path = "$user","public","heroku_ext"
```

**Comment by gaganawhad** [comment]
@jeremyevans, thanks for that quick response. I appreciate it. 

While we are not using Rails, I do see that the schema dumper was picked from `sequel-rails` plugin. I'll connect with them. Thanks!
